File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IProxyManagerView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.ProxyConfig;
import dev.ragnarok.fenrir.mvp.core.IMvpView;


public interface IProxyManagerView extends IMvpView, IErrorView {
    void displayData(List<ProxyConfig> configs, ProxyConfig active);

    void notifyItemAdded(int position);

    void notifyItemRemoved(int position);

    void setActiveAndNotifyDataSetChanged(ProxyConfig config);

    void goToAddingScreen();
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IProxyManagerView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.ProxyConfig
import dev.ragnarok.fenrir.mvp.core.IMvpView

interface IProxyManagerView : IMvpView, IErrorView {
    fun displayData(configs: MutableList<ProxyConfig>, active: ProxyConfig?)
    fun notifyItemAdded(position: Int)
    fun notifyItemRemoved(position: Int)
    fun setActiveAndNotifyDataSetChanged(config: ProxyConfig?)
    fun goToAddingScreen()
}